gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/TokenServiceImpl.java

```diff
@@ -153,6 +153,19 @@ public Token findByToken(String token) {
         }
     }
 
+    @Override
+    public boolean tokenExistsForUser(String tokenId, String userId) {
+        try {
+            Optional<Token> byId = tokenRepository.findById(tokenId);
+            return byId.map(token -> token.getReferenceId().equals(userId)).orElse(false);
```

Comment:
You can leave it as it is, but for me, using a filter makes it more readable 
byId.map(Token::getReferenceId)
          .filter(ref -> ref.equals(userId))
          .isPresent();